src/plugin/scope.ml

```diff
@@ -231,10 +231,11 @@ module Type_tree = struct
           |> add_names ~path ~ocaml_name map
         in
 
+        (* Prevent uncapitalizing to make correct rpc name paths *)
         let map =
           create_name_map
-            ~standard_f:(Names.field_name ~mangle_f:(fun x -> x))
-            ~mangle_f:(Names.field_name ~mangle_f)
+            ~standard_f:(Names.field_name ~uncapitalize:false ~mangle_f:(fun x -> x))
```

Comment:
I think there is a problem here!
I assume the reason for this change is that the name of the Rpc type module is capitalized, so it might clash with other names. 

The map is needed to detect name clashes caused by altering the character of the name. As the Rpc type module is capitalized, we might introduce a multiple name clashes.

I suggest that you add a test with a simple service interface, e.g.
```protobuf
service Test {
  Rpc Call (Request) return (Reply);
  Rpc call (Request) return (Reply);
  Rpc CALL (Request) return (Reply);
  Rpc cALL (Request) return (Reply);
}

and make sure that the names are mapped correctly and does not produce a clash. 

```

Reply:
Sure. Some strategies to avoid name clash can be thought:
1. add prefix when the (uncapitalized) name clash (e.g. `call_1`, `call_2`, ...)
2. add suffix (e.g. `_call`, `_Call`)
3. throw error

Which strategy do you prefer, or dou you have other way?

Reply:
In the simple case - no name clash - the system should produce the most natural mapping, just like your example at the top of the PR. (`module Call = struct ... end`  and `let call = ... `). We should record names used within the scope and if a name clash is encountered, the system should apply methods to ensure use of unique names. 

My suggestion would be to have capture all names emitted in a module based on a service. Whenever a new name is to be emitted to this scope, call a function which will ensure no name clashes. 

Something like:
```ocaml
module Local_scope = struct
  type t = (string, unit) Hashtbl.t
  let init () = Hashtbl.create 2
  let get_unique_name t preferred_name = 
    let rec inner name = 
      match Hashtbl.mem t name with
      | false -> name
      | true -> inner (name ^ "'")
   in 
   let name = inner preferred_name in
   Hashtbl.add t name ();
   name
end
```

Whenever we try to create a name inside the service module, we map the name through `Local_scope.get_unique_name`